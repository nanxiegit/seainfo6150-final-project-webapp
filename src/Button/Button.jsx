import React from "react";
import PropTypes from "prop-types";

import styles from "./Button.module.css";

const Button = (props) => {
  return (
    <button className={styles.button} onClick={props.onClick}>
      {props.buttonText}
    </button>
  );
};

Button.propTypes = {
  buttonText: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
export default Button;
