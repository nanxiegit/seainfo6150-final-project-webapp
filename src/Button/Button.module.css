.button {
    background: rgb(88, 88, 88);
    border-radius: 10px;
    color: #fff;
    padding: 10px;
    font-family: fantasy;
    font-weight: bold;
    font-size: 14px;
}
