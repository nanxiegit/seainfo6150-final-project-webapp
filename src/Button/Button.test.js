import React from "react";
import Button from "./Button";

describe("Button tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Button  buttonText={ "sip off" } onClick={() => {}} 
    />);
    expect(container).toMatchSnapshot();
  });
});