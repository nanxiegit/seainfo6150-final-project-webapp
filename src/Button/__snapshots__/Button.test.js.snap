// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests renders correctly 1`] = `
<div>
  <button
    class="button"
  >
    sip off
  </button>
</div>
`;
